dataproc/Dockerfile

```diff
@@ -1,29 +1,72 @@
-FROM ubuntu:20.04
+FROM ubuntu:22.04
+
+# Give this Dockerfile the ability to checkout a specific commit hash. This is
+# useful if you want to build a specific Hail version inside the image instead
+# of the latest commit on master.
+ARG COMMIT_HASH
```

Comment:
`HAIL_COMMIT` might be a better name, to emphasise that the repo it's talking about is hail, and it doesn't need to be a hash — e.g., could well be a tag.

What is the relationship between this and `HAIL_VERSION` in the README? I think they'd be mostly the same thing.

Am I missing something, or how would you specify these ARGs if building via the `gcloud builds submit` command shown in the README?

Reply:
Oh, I have completely forgotten to update the README inside `./dataproc` with instructions on how to use the new build args. `HAIL_COMMIT` sounds like a better name as it captures things that aren't just commit hashes.

As far as I can tell, `HAIL_VERSION` is only used as a docker tag, for example I tried setting `HAIL_VERSION=0.2.105` and building the image from the latest commit, hoping it would build 0.2.105 but it built the latest version that HEAD was at despite this. I will update the README.

Reply:
I've renamed to HAIL_COMMIT and will prepare a config.